editor.planx.uk/src/pages/FlowEditor/components/Sidebar/Publish/PublishFlowButton.tsx

```diff
@@ -96,14 +96,14 @@ export const PublishFlowButton: React.FC<{ previewURL: string }> = ({
     const user = await lastPublisher(flowId);
 
     setLastPublishedTitle(formatLastPublishMessage(date, user));
-  });
+  }, [flowId]);
 
   const _validateAndDiffRequest = useAsync(async () => {
     const newChanges = await validateAndDiffFlow(flowId);
     setAlteredNodes(
       newChanges?.data.alteredNodes ? newChanges.data.alteredNodes : [],
     );
-  });
+  }, [flowId]);
```

Comment:
_Technically_ this should update each time there's an operation, but I think this would be calling this expensive API call a lot more than we need to.

Having it update once when navigating into a flow feels fine as a fix for now.

Reply:
Neat :)